From 8d2acef7c5d7179db19b3c8a0ae9ae4c152faaff Mon Sep 17 00:00:00 2001
From: Andy Green <andy@openmoko.com>
Date: Sun, 10 Aug 2008 09:16:04 +0100
Subject: [PATCH] fix-no-discharging.patch

We failed to report status of "discharging", instead reporting
"not charging" even if we knew that the charger was not present.

This patch corrects it and reports "discharging" when charger
is absent.

Signed-off-by: Andy Green <andy@openmoko.com>
---
 drivers/power/bq27000_battery.c |   10 +++++++++-
 1 files changed, 9 insertions(+), 1 deletions(-)

diff --git a/drivers/power/bq27000_battery.c b/drivers/power/bq27000_battery.c
index 4c7e862..cd4968e 100644
--- a/drivers/power/bq27000_battery.c
+++ b/drivers/power/bq27000_battery.c
@@ -187,9 +187,17 @@ static int bq27000_battery_get_property(struct power_supply *psy,
 				val->intval = POWER_SUPPLY_STATUS_CHARGING;
 				break;
 			}
+			val->intval = POWER_SUPPLY_STATUS_NOT_CHARGING;
+			break;
 		}
 
-		val->intval = POWER_SUPPLY_STATUS_NOT_CHARGING;
+		/*
+		 * platform provided definite indication of charger presence,
+		 * and it is telling us it isn't there... but we are on so we
+		 * must be running from battery --->
+		 */
+
+		val->intval = POWER_SUPPLY_STATUS_DISCHARGING;
 		break;
 
 use_bat:
-- 
1.5.6.3

